src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import Downshift from 'downshift';
+import PropTypes from 'prop-types';
+
+import { InputGroup, Button } from '../../index';
+import AutoCompleteInput from './AutoCompleteInput';
+import AutoCompleteItems from './AutoCompleteItems';
+import { KEY_CODES } from '../../common/helpers';
+
+export const getActiveItems = items =>
+  items
+    .filter(
+      ({ disabled, type }) => !disabled && !['header', 'divider'].includes(type)
+    )
+    .map(({ text }) => text);
+
+class AutoComplete extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      inputValue: ''
+    };
+  }
+
+  render() {
+    const {
+      onSearch,
+      labelText,
+      onInputUpdate,
+      items,
+      actionText,
+      ...rest
+    } = this.props;
+
+    const activeItems = getActiveItems(items);
+
+    return (
+      <Downshift
+        onStateChange={({ inputValue, ...rest }) => {
```

Comment:
Let's try to keep this method lean.
Can you move this method into `this.handleStateChange`?

Reply:
I like abstracting like that when possible. The extra challenge here is needed the `onInputUpdate` function in the scope. We could forward it through by currying or something but I think it's easier to see inline.